crypto/bls12381/bls12_381.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2020 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package bls12381
+
+func Init() {
+	cfgArch()
+}
+
+/*
+	Field Constants
+*/
+
+var modulus fe = fe{0xb9feffffffffaaab, 0x1eabfffeb153ffff, 0x6730d2a0f6b0f624, 0x64774b84f38512bf, 0x4b1ba7b6434bacd7, 0x1a0111ea397fe69a}
```

Comment:
Any particular reason for keeping this as non-pointer and having all the other constants as pointers? There's also no reason to define the type, implicit is enough.

Reply:
Type definition is removed. 

From the perspective of assembly coding being non pointer of this data allows us to use its limbs directly without assigning its address to a general purpose register. So that, we save one more register space for more efficient multiplication processing. You can see same trick with bn254 curve implementation as well.

Reply:
Thx for the explanation